test/test_meta.py

```diff
@@ -1167,6 +1165,8 @@ def _run_dispatch_meta_test(self, device, dtype, op, symbolic_meta, inplace, all
             func = op.get_inplace()
             if not func:
                 self.skipTest("No inplace variable for this op")
+            if op.promotes_int_to_float and not dtype.is_floating_point:
+                self.skipTest("Op promotes to float, which is impossible for inplace with non-float input")
```

Comment:
Let's test here that indeed it promotes int to float.

Reply:
I thought about, but not sure we should for two reasons:

1. If we have such a check, we should probably run it on all sample inputs. Sure, running it on least one already gives a good indicator, but no real assurances.
2. The test would then check if the user put the right configuration in the `OpInfo`. While its not bad to have such a test to avoid mistakes, this probably shouldn't be inside the meta dispatch tests?